### PR TITLE
Add more ITimeZoneService related commands

### PR DESCRIPTION
**Pull request by Thog**
This PR adds timeGetDeviceLocationName, timeSetDeviceLocationName, timeGetTotalLocationNameCount, timeLoadLocationNameList, timeLoadTimeZoneRule, timeToPosixTime, timeToPosixTimeWithMyRule, and timeToCalendarTime.

### CODE REVIEW

File: nx/include/switch/services/time.h [hunk 1]
```diff
@@ -35,6 +35,15 @@ typedef struct {
     s32 offset; ///< Seconds relative to UTC for this timezone.
 } TimeCalendarAdditionalInfo;
 
+typedef struct {
+    char reserved[0x4000];
+} TimeZoneRule;
+
+typedef struct
+{
```
Comment (fincs) [inline nx/include/switch/services/time.h]: `typedef struct {`

File: nx/include/switch/services/time.h [hunk 2]
```diff
@@ -35,6 +35,15 @@ typedef struct {
...
+    char name[0x24];
+} LocationName;
```
Comment (fincs) [inline nx/include/switch/services/time.h]: `TimeLocationName`

File: nx/include/switch/services/time.h [hunk 3]
```diff
@@ -35,6 +35,15 @@ typedef struct {
     s32 offset; ///< Seconds relative to UTC for this timezone.
 } TimeCalendarAdditionalInfo;
 
+typedef struct {
+    char reserved[0x4000];
```
Comment (yellows8) [inline nx/include/switch/services/time.h]: u8 data

File: nx/source/services/time.c [hunk 4]
```diff
@@ -191,6 +191,222 @@ Result timeSetCurrentTime(TimeType type, u64 timestamp) {
     return rc;
 }
 
+Result timeGetDeviceLocationName(LocationName *name)
+{
```
Comment (yellows8) [inline nx/source/services/time.c]: Move to previous line, same below.

File: nx/include/switch/services/time.h [hunk 5]
```diff
@@ -50,5 +59,15 @@ Result timeGetCurrentTime(TimeType type, u64 *timestamp);
...
 Result timeSetCurrentTime(TimeType type, u64 timestamp);
 
+Result timeGetDeviceLocationName(LocationName *name);
+Result timeSetDeviceLocationName(const LocationName *name);
+Result timeGetTotalLocationNameCount(u32 *total_location_name_count);
+Result timeLoadLocationNameList(u32 index, LocationName *location_name_array, size_t location_name_size, u32 *location_name_count);
+
+Result timeLoadTimeZoneRule(const LocationName *name, TimeZoneRule *rule);
+
+Result timeToPosixTime(const TimeZoneRule *rule, TimeCalendarTime caltime, u64 *timestamp_list, size_t timestamp_list_size, u32 *timestamp_count);
```
Comment (yellows8) [inline nx/include/switch/services/time.h]: Use ptr for caltime, same below.

**Pull request merged**